src/platforms/linux/daemon/dbusservice.cpp

```diff
@@ -40,6 +41,13 @@ DBusService::DBusService(QObject* parent) : Daemon(parent) {
 
 DBusService::~DBusService() { MVPN_COUNT_DTOR(DBusService); }
 
+WireguardUtilsLinux* DBusService::wgutils() {
+  if (!m_wgutils) {
+    m_wgutils = new WireguardUtilsLinux();
```

Comment:
This is a bug: here you are allocating an object but you do not deallocate it.
The simplest solution is to set a parent to WireguardUtilsLinux and pass it to QObject. See this: https://doc.qt.io/qt-5/qobject.html#QObject "The destructor of a parent object destroys all child objects."
\\

Reply:
If you add MVPN_COUNT_CTOR/DTOR in the wgutils, you will see that in debug builds the leak is reported.

Reply:
Thank you. I'll fix this. Sorry I still have things to learn to get this ingrained in my head.

Reply:
Resolved in 9761274